src/provider/store/Store.ts

```diff
@@ -37,7 +44,8 @@ export class Store {
         this._store.dispatch(action);
     }
 
-    public watchForChange<T>(getObject: (state: RootState) => T, observer: StoreChangeObserver<T>): Unsubscribe {
+    public async watchForChange<T>(getObject: (state: RootState) => T, observer: StoreChangeObserver<T>): Promise<Unsubscribe> {
```

Comment:
There's two places that call this that are potentially affected by this becoming `async` - `subscribe` in `NotificationCenter`, and `subscribe` in `ToastManager`. The first is async but returns immediately rather than `await`ing this call. The second is synchronous. Are both intentional?

Reply:
Cool thanks.  They should be fire and forget so neither need to be in the async.

Reply:
I have fixed the methods to remove the async where this was being used.